jdaviz/core/config.py

```diff
@@ -0,0 +1,100 @@
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Filename: config.py
+# Project: core
+# Author: Brian Cherinka
+# Created: Tuesday, 23rd June 2020 1:22:32 pm
+# License: BSD 3-clause "New" or "Revised" License
+# Copyright (c) 2020 Brian Cherinka
+# Last Modified: Tuesday, 23rd June 2020 1:22:32 pm
+# Modified By: Brian Cherinka
+
+#
+#  This file contains helper function related to configuration handling
+#
+
+from __future__ import print_function, division, absolute_import
+import copy
+import os
+import pathlib
+import yaml
+
+
+def read_configuration(path=None):
+    """ Loads a configuration from a YAML file
+
+    Loads application configuration settings from a YAML file
+
+    Parameters
+    ----------
+    path : str, optional
+        Path to the configuration file to be loaded. If None, loads the
+        default configuration.
+
+    Returns:
+        A dictionary object
+    """
+    if not isinstance(path, (str, type(None))):
+        raise ValueError('path must be a string')
+
+    # Parse the default configuration file
+    default_path = pathlib.Path(__file__).resolve().parent.parent / "configs"
+
+    if path is None or path == 'default':
+        path = default_path / "default" / "default.yaml"
+    elif path == 'cubeviz':
+        path = default_path / "cubeviz" / "cubeviz.yaml"
+    elif path == 'specviz':
+        path = default_path / "specviz" / "specviz.yaml"
+    elif path == 'mosviz':
+        path = default_path / "mosviz" / "mosviz.yaml"
+    elif not os.path.isfile(path):
+        raise ValueError("Configuration must be path to a .yaml file.")
+
+    with open(path, 'r') as f:
+        config = yaml.safe_load(f)
+
+    return config
+
+
+def get_configuration(path=None, section=None, config=None):
+    """ Returns a copy of the application configuration
+
+    Returns a copy of the configuration specification.  If path
+    is not specified, returns the currently loaded configuration.
```

Comment:
The description here isn't totally accurate, the current configuration retrieval is only in `app.get_configuration`. Seems like the organization of what goes in which `get_configuration` could be a bit more elegant/logically organized but then again I'm not exactly sure how to switch it up.

Reply:
@rosteen Yeah the docstring here might not be updated.  This code was originally in `app.get_configuration` before I split it out.  I needed to split it out so I could access configurations outside the app context.  The idea behind the new `app.get_configuration` was by default to return the currently loaded one, otherwise return the config specified in the input.